MH4U Dex/CoreDataController.m

```diff
@@ -517,28 +517,61 @@ - (NSManagedObjectModel *)managedObjectModel
 
 - (NSPersistentStoreCoordinator *)persistentStoreCoordinator
 {
-    // The persistent store coordinator for the application. This implementation creates and return a coordinator, having added the store for the application to it.
     if (_persistentStoreCoordinator != nil) {
         return _persistentStoreCoordinator;
     }
     
-    // Create the coordinator and store
-    
-    _persistentStoreCoordinator = [[NSPersistentStoreCoordinator alloc] initWithManagedObjectModel:[self managedObjectModel]];
-    NSURL *storeURL = [[self applicationDocumentsDirectory] URLByAppendingPathComponent:@"MH4U_Dex.sqlite"];
-    NSError *error = nil;
-    NSString *failureReason = @"There was an error creating or loading the application's saved data.";
-    if (![_persistentStoreCoordinator addPersistentStoreWithType:NSSQLiteStoreType configuration:nil URL:storeURL options:nil error:&error]) {
-        // TODO: fix all of this later.
-        // Report any error we got.
-        NSDictionary *dict = @{
-                               NSLocalizedDescriptionKey: @"Failed to initialize the application's saved data",
-                               NSLocalizedFailureReasonErrorKey: failureReason,
-                               NSUnderlyingErrorKey: error,
-                               };
-        error = [NSError errorWithDomain:@"YOUR_ERROR_DOMAIN" code:9999 userInfo:dict];
-        // Replace this with code to handle the error appropriately.
-        // abort() causes the application to generate a crash log and terminate. You should not use this function in a shipping application, although it may be useful during development.
+    NSURL *storeURL = [self.applicationDocumentsDirectory URLByAppendingPathComponent:@"MH4U_DEX.sqlite"];
+    /*
+     Set up the store.
+     For the sake of illustration, provide a pre-populated default store.
+     */
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    // If the expected store doesn't exist, copy the default store.
+    if (![fileManager fileExistsAtPath:storeURL.path]) {
+        // typically the main store name is 'appName.sqlite'
+        NSURL *defaultStoreURL = [[NSBundle mainBundle] URLForResource:@"PREBUILT_STORE" withExtension:@"sqlite"];
+        if (defaultStoreURL) {
```

Comment:
if it is null, then the file doesn't exist to be copied.  I guess it wouldn't break anything if I removed that check, since copyItemAtURL just returns NO if it fails